addons/src/ao_markword.c

```diff
@@ -190,6 +208,14 @@ void ao_mark_document_open(AoMarkWord *mw, GeanyDocument *document)
 		FALSE,
 		G_CALLBACK(on_editor_button_press_event),
 		mw);
+		
+	plugin_signal_connect(
```

Comment:
This is the wrong place for the signal connect.
Here we connect to the signal on each opened document but never disconnect from it.

It would be better to re-use the already existing signal callback `ao_editor_notify_cb` in addons.c and add there a call to `on_editor_notify` after making it public.